src/NuGet.Core/NuGet.ProjectModel/CacheFile.cs

```diff
@@ -23,6 +23,7 @@ public class CacheFile : IEquatable<CacheFile>
         /// </summary>
         public IList<string> ExpectedPackageFilePaths { get; set; }
 
+        [Obsolete("File existence checks are a function of time not the cache file content.")]
```

Comment:
Should we change this from an auto-prop, to a property whose getting and setting method bodies throw `NotImplementedException` or `NotSupportedException`.

```cs
[Obsolete("This API is no longer supported")]
public bool HasAnyMissingPackageFiles
{
  get => throw new NotImplementedException("This API is no longer supported");
  set => throw new NotImplementedExeption("This API is no longer supported");
}
```

Imagine a 3rd party PkgX that uses NuGet.ProjectModel, and this `HasAnyMissingPackageFiles` property. If I create a project that uses PkgX, and also directly reference a newer version of NuGet.ProjectModel, then from my perspective PkgX will have a bug, because the behaviour is wrong, but I'll never see this obsolete compiler warning because I'm not referencing this API directly. At least a not implemented exception will prevent silent unexpected behaviour.

Although, changing the API to always throw, allows us to give an easy to understand error message, but functionally is effectively equivalent to deleting the public API. Whether we throw or delete, any code using the API will be broken and anyone using the package, directly or indirectly, will have no choice but to make changes. I can't decide if that's better or worse than another app potentially calling the API, but getting unexpected behaviour.

If we want to do the right thing from an API compatibility point of view, we can use a `Lazy`, or equivalent, to only hit the filesystem if someone actually uses this API. I guess it's a question of how much effort we want to go to for a public API that's unlikely to be used by anyone. On GitHub, I can only find this API in forks of NuGet.Client.

Anyway, my comment is more of "something to think about", rather than a request for a change.

Reply:
I like your idea for throwing. I'll push a change for that and resolve the comment when I do that.